### ISSUE TITLE
SEACASAprepro_lib_aprepro_lib_unit_test Not Run

### ISSUE THREAD

**Issue by ibaned** [trilinos/Trilinos#1482]
Hello, I was running the `checkin_test_sems` script for a very trivial change and it failed because the test `SEACASAprepro_lib_aprepro_lib_unit_test` did not run. Any ideas why that might be? I made no CMake changes whatsoever, only MueLu source files changed (see #1479 )

```
FAILED: Trilinos/MPI_RELEASE_DEBUG_SHARED_PT: passed=422,notpassed=1

Fri Jul  7 18:17:11 MDT 2017

Enabled Packages: MueLu
Disabled Packages: PyTrilinos,Claps,TriKota
Enabled all Forward Packages
Hostname: ceerws1113
Source Dir: /scratch/daibane/Trilinos/cmake/tribits/ci_support/../../..
Build Dir: /scratch/daibane/Trilinos/CHECKIN/MPI_RELEASE_DEBUG_SHARED_PT

CMake Cache Varibles: -DTrilinos_TRIBITS_DIR:PATH=/scratch/daibane/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=BASIC -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=300.0 -DBUILD_SHARED_LIBS=ON -DTrilinos_DISABLE_ENABLED_FORWARD_DEP_PACKAGES=ON -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_SECONDARY_TESTED_CODE:BOOL=OFF -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/MpiReleaseDebugSharedPtSettings.cmake,cmake/std/BasicCiTestingSettings.cmake -DTrilinos_ENABLE_MueLu:BOOL=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=ON -DTrilinos_ENABLE_PyTrilinos:BOOL=OFF -DTrilinos_ENABLE_Claps:BOOL=OFF -DTrilinos_ENABLE_TriKota:BOOL=OFF
Make Options: -j10
CTest Options: -j5

Pull: Passed (0.00 min)
Configure: Passed (1.33 min)
Build: Passed (104.14 min)
Test: FAILED (18.63 min)

99% tests passed, 1 tests failed out of 423

Label Time Summary:
MueLu                = 327.52 sec (56 tests)
Panzer               = 307.62 sec (129 tests)
Piro                 =  31.64 sec (12 tests)
ROL                  = 1007.75 sec (133 tests)
SEACAS               =   0.21 sec (2 tests)
Stokhos              = 107.68 sec (75 tests)
TrilinosCouplings    =  47.16 sec (19 tests)

Total Test time (real) = 1117.85 sec

The following tests FAILED:
	  1 - SEACASAprepro_lib_aprepro_lib_unit_test (Not Run)
Errors while running CTest

Total time for MPI_RELEASE_DEBUG_SHARED_PT = 124.11 min
```

@trilinos/framework @trilinos/seacas

**Comment by bartlettroscoe** [comment]
@ibaned, can you please attach the `checkin-test.out` log file?

**Comment by ibaned** [comment]
Here it is
[checkin-test.out.txt](https://github.com/trilinos/Trilinos/files/1136236/checkin-test.out.txt)

**Comment by bartlettroscoe** [comment]
Looks like the checkin-test.py script is working correctly as it is only enabling MueLu as shown by:

```

Modified file: 'packages/muelu/adapters/xpetra/MueLu_CreateXpetraPreconditioner.hpp'
  => Enabling 'MueLu'!

Full package enable list: [MueLu]

...

cmakePkgOptions: [u'-DTrilinos_ENABLE_MueLu:BOOL=ON', '-DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON', '-DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=ON', '-DTrilinos_ENABLE_PyTrilinos:BOOL=OFF', '-DTrilinos_ENABLE_Claps:BOOL=OFF', '-DTrilinos_ENABLE_TriKota:BOOL=OFF']
```

Can you please attach the file MPI_RELEASE_DEBUG_SHARED_PT/configure.out?

Hopefully that will show why this test is getting enabled.

**Comment by bartlettroscoe** [comment]
I think I know why this is happening.  The tests in:

```
 packages/seacas/applications/aprepro/CMakeLists.txt
```

are not being protected by `${PACKAGE_NAME}_ENABLE_TESTS` (or put into a test subdir which is protected).

I will take a look at how to fix this and suggest a patch to the native SEACAS repo (and then fix it natively in Trilinos).

**Comment by bartlettroscoe** [comment]
Actually, I will just update `TRIBITS_ADD_TEST()` and `TRIBITS_ADD_ADVANCED_TEST()` with an if statement for `${PACKAGE_NAME}_ENABLE_TESTS` internally if testing not enabled for that package.  That will fix cases like this once and for all.  I will try to get this pushed to TriBITS then Trilinos later today.

**Comment by mhoemmen** [comment]
@trilinos/seacas

**Comment by ibaned** [comment]
I almost missed the request for `configure.out`, here is it
[configure.out.txt](https://github.com/trilinos/Trilinos/files/1136416/configure.out.txt)

**Comment by bartlettroscoe** [comment]
@ibaned, it does not look like this configure.out file is from the same run.  That file shows that all package are being enabled:

```
Enabling all SE packages that are not currently disabled because of Trilinos_ENABLE_ALL_PACKAGES=ON ...
```

Anyway, I will look into this and fix as described above.

**Comment by ibaned** [comment]
@bartlettroscoe I don't see that line in that file...

**Comment by bartlettroscoe** [comment]
The fix is being pushed right now with checkin-test-sems.sh (pushing TriBITSPub/TriBITS#204).

NOTE: This did not break the post-push CI server because it specifically only tests the package being processed, and therefore ignores upstream packages.  But even if it did run all tests, those SEACAS tests would pass because the exectuables were built when SEACAS was processed in the first CI build.

And the checkin-test-sems.sh script would have passed the changes to SEACAS since SEACAS was being changed and its tests would have been enabled.  So even if Brent had run the checkin-test-sems.sh script to push the SEACAS changes, it would have allowed the push.

**Comment by bartlettroscoe** [comment]
Should be fixed as of commit 9a5a5e4f546ee pushed last night.   Now the CI build shows:

```
Processing enabled package: SEACAS (Exodus, Ioss, Aprepro_lib)
-- SEACASAprepro_lib_aprepro_lib_unit_test: NOT added test because SEACASAprepro_lib_ENABLE_TESTS=', SEACAS_ENABLE_TESTS='''.
-- SEACASAprepro_lib_aprepro_lib_array_test: NOT added test because SEACASAprepro_lib_ENABLE_TESTS=', SEACAS_ENABLE_TESTS='''.
```

when SEACAS is libs are enabled but not test.

@ibaned, please let me know if fixes your usage of the checkin-test-sems.sh script.

Putting into review.

**Comment by ibaned** [comment]
@bartlettroscoe Yes! The `checking-test-sems.sh` script just succeeded. This is resolved for me. Thank you!
